analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,329 @@
+  dplyr::filter(ID %in% lgat_wgs_biospecimen_ids)
+```
+
+### Overlap tandem duplication in FGFR1 with exons 10-18 which make up the kinase domain, per [comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/790#issuecomment-729229244).
+
+First, create genomic ranges objects for CNV seg file and gene.
+
+```{r}
+# Genomic range for cnv seg file
+cnv_gr <- cnv %>%
+  # format
+  dplyr::rename(
+    chr = chrom, start = loc.start, end = loc.end,
+    copy_number = copy.num
+  ) %>%
+  dplyr::select(-num.mark, -seg.mean) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE
+  )
+
+# Genomic range for gene location and domain overlap file
+domain_gr <- bioMartDataPfamFGFR1 %>%
+  # keep only domain info
+  dplyr::filter(!is.na(domain_start), !is.na(domain_end)) %>%
+  # format
+  dplyr::mutate(
+    strand = if_else(strand == "-1", "-", "+"),
+    domain_chr = paste0("chr", domain_chr)
+  ) %>%
+  dplyr::rename(chr = domain_chr, start = domain_start, end = domain_end) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE,
+    seqnames.field = "chr"
+  )
+```
+
+
+
+### Distribution of manta SV
+SV should be annotated as tandem duplicate and overlap 
+"Protein tyrosine kinase" domain of FGFR1 in biospecimens
+
+```{r}
+
+# Genomic range for manta file
+manta_gr <- manta_lgat %>%
+  # format
+  dplyr::rename(
+    chr = SV.chrom, start = SV.start, end = SV.end,
+    ALT = ALT
+  ) %>%
+  # format with chr to match to cnv
+  dplyr::mutate(chr = paste0("chr", chr)) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE
+  )
+
+# overlap manta and domain
+overlaps_manta <- IRanges::subsetByOverlaps(manta_gr, domain_gr)
+
+# filter for fgfr only overlap
+overlaps_manta_fgfr <- overlaps_manta[which(overlaps_manta$Gene.name == "FGFR1" &
+  # filter for tandem duplication
+  overlaps_manta$ALT == "<DUP:TANDEM>"), ]
+
+overlaps_manta_fgfr_df <- data.frame(
+  biospecimen_id = overlaps_manta_fgfr$Kids.First.Biospecimen.ID.Tumor,
+  ALT = overlaps_manta_fgfr$ALT,
+  # annnotation for dup tandem
+  FGFR_DUP_TANDEM = "Yes",
+  stringsAsFactors = FALSE
+)
+
+overlaps_manta_fgfr_df
+```
+4 biospecimens have tandem duplication that overlap tyrosine kinase domain in FGFR1
+
+
+### Distribution of copy number
+CNV that are within FGFR1 gene region only and overlap "Protein tyrosine kinase" domain in FGFR1 in biospecimens
+
+```{r}
+
+# gene region overlap
+fgfr_gene <- GenomicRanges::GRanges(
+  seqnames = paste0("chr", domain_gr$chromosome_name),
+  ranges = IRanges::IRanges(start = domain_gr$gene_start, end = domain_gr$gene_end)
+)
+
+# subset cnv+domain grange to be within FGFR1 gene region
+cnv_fgfr_gr <- IRanges::subsetByOverlaps(cnv_gr, fgfr_gene, type = "within", maxgap = )
```

Comment:
In the context of `within`: `maxgap... is the maximum amount by which the subject may be wider than the query.` So is there an extent to which the CNV repeated region would be so large that we should treat it differently? If yes, then we should set maxgap, if no, then I think we can leave off maxgap.

Reply:
ah yes this was a typo from testing. I've implemented a type="within" and minOverlap= domain_gr length which I think will also make this a 1 step subsetByOverlaps command.

Reply:
I think this sounds resolved, right @cansavvy ?

Reply:
Yes. Sounds good.